tests/ludwig/features/test_category_feature.py

```diff
@@ -40,10 +43,12 @@ def test_category_input_feature(
     category_def[ENCODER][TYPE] = encoder
 
     # pickup any other missing parameters
-    CategoryInputFeature.populate_defaults(category_def)
+    defaults = CategoryInputFeatureConfig().to_dict()
```

Comment:
Smaller/separate PR suggestion: Removing and replacing `populate_defaults` with the config object the could be its own separate change.

Reply:
Agreed! In this case however, the removing of `populate_defaults` went along with the full removal of `merge_with_defaults` which would take quite a while to decouple into a separate PR, so for this I would opt to keep them together!